src/devices/Hcsr04/Hcsr04.cs

```diff
@@ -40,32 +42,31 @@ public Sonar(int triggerPin, int echoPin)
         /// <summary>
         /// Gets the current distance in cm.
         /// </summary>
-        public double GetDistance()
+        private double GetDistance()
         {
             _timer.Reset();
 
             // Trigger input for 10uS to start ranging
             // ref https://components101.com/sites/default/files/component_datasheet/HCSR04%20Datasheet.pdf
+            _controller.Write(_trigger, PinValue.Low);
+            Thread.Sleep(TimeSpan.FromMilliseconds(0.002));
             _controller.Write(_trigger, PinValue.High);
             Thread.Sleep(TimeSpan.FromMilliseconds(0.01));
             _controller.Write(_trigger, PinValue.Low);
             
-            _timer.Start();
-            
             while(_controller.Read(_echo) == PinValue.Low)
             {
             }
-
+            _timer.Start();
             while(_controller.Read(_echo) == PinValue.High)
             {
             }
-
             _timer.Stop();
             
             TimeSpan elapsed = _timer.Elapsed;
 
             // distance = (time / 2) × velocity of sound (34300 cm/s)
-            return (double)(elapsed.TotalMilliseconds * 34.3) / 2 - 8;
+            return elapsed.TotalMilliseconds / 2.0 * 34.3;
```

Comment:
`- 8` was based on the experiments. Any reasons to remove?

Reply:
@krwq When I first saw this formula, I was wondering why this needs to -8, and it's a cm. I think about the thickness of the sensor, but it's not very reliable. Ultimately, it's still a mistake in measuring time. You can try it again with new code.😁